src/commands/up.rs

```diff
@@ -194,6 +203,23 @@ impl UpCommand {
             .load_resolved_app_source(resolved_app_source, &working_dir)
             .await?;
 
+        if let Some(components) = &self.components {
+            // Create a temporary app to access parsed component and trigger information
+            let tmp_app = spin_app::App::new("tmp", locked_app.clone());
+            let component_trigger_id_map = component_trigger_map(&tmp_app, components)?;
+            retain_components(&mut locked_app, component_trigger_id_map);
```

Comment:
Can these top two lines move into `retain_components`? i.e.

```suggestion
            retain_components(&mut locked_app, components);
```